src/Scenario/Merger.php

```diff
@@ -0,0 +1,248 @@
+            }
+
+            $argumentName = $argument['@name'];
+            $argumentType = $argument['@xsi:type'];
+
+            if ($argumentType !== self::XSI_TYPE_ARRAY) {
+                throw new ValidationException(sprintf(
+                    'xsi:type "%s" not allowed in argument "%s"',
+                    $argumentType,
+                    $argumentName
+                ));
+            }
+
+            $arguments[] = [
+                'name' => $argumentName,
+                'xsi:type' => $argumentType,
+                'items' => $this->collectItems(
+                    $argument['item'] ?: []
+                )
+            ];
+        }
+
+        $stepData = [
+            'name' => $stepName,
+            'type' => $step['@type'],
+            'arguments' => $arguments
+        ];
+
+        return $stepData;
+    }
+
+    /**
+     * Collect scenario data
+     *
+     * @param string $scenario
+     * @return array
+     * @throws ValidationException
+     */
+    private function collectScenario(string $scenario): array
+    {
+        $scenarioPath = $this->systemList->getRoot() . '/' . $scenario;
+
+        if (!$this->file->isExists($scenarioPath)) {
+            $scenarioPath = $this->systemList->getMagentoRoot() . '/' . $scenario;
+        }
+
+        if (!$this->file->isExists($scenarioPath)) {
+            throw new ValidationException(sprintf(
+                'Scenario %s does not exist',
+                $scenario
+            ));
+        }
+
+        try {
+            return $this->encoder->decode(
+                $this->file->fileGetContents($scenarioPath),
+                XmlEncoder::FORMAT,
+                [
+                    XmlEncoder::AS_COLLECTION => true,
+                    XmlEncoder::ROOT_NODE_NAME => self::ROOT_NODE,
+                ]
+            ) ?: [];
+        } catch (FileSystemException $exception) {
+            throw new ValidationException($exception->getMessage(), $exception->getCode(), $exception);
+        }
+    }
+
+    /**
+     * Recursively collect items
+     *
+     * @param array $items
+     * @return array
+     * @throws ValidationException
+     */
+    private function collectItems(array $items): array
+    {
+        $newItems = [];
+
+        foreach ($items as $item) {
+            if (!is_array($item)) {
+                throw new ValidationException('Wrong formatted item provided');
+            }
+
+            $itemName = $item['@name'] ?? '';
+
+            if ($missedArgs = array_diff(self::$itemRequiredArgs, array_keys($item))) {
+                throw new ValidationException(sprintf(
+                    'Argument(s) "%s" are missed from item "%s"',
+                    implode(', ', $missedArgs),
+                    $itemName
+                ));
+            }
+
+            $newItem = [
+                'name' => $itemName,
+                'xsi:type' => $item['@xsi:type'],
+            ];
+
+            if (isset($item['#'])) {
```

Comment:
What is the meaning of `#`? It appears to indicate a leaf node in the item tree, but why? Where does it come from? A brief comment or replacing `'#'` with a class constant would help for clarity

Reply:
Added constant